chrome/content/zotero/preferences/preferences_sync.js

```diff
@@ -177,6 +177,129 @@ Zotero_Preferences.Sync = {
 		this.displayFields();
 		yield Zotero.Sync.Runner.deleteAPIKey();
 	}),
+	
+	
+	showSyncedLibrariesDialog: function() {
+		var io = {};
+		window.openDialog('chrome://zotero/content/preferences/syncedLibraries.xul',
+			"zotero-preferences-syncedLibrariesDialog", "chrome,modal,centerscreen", io);
+	},
+	
+	
+	dblClickSyncedLibrary: function (index) {
+		if (index < 3) {
+			return;
+		}
+		var treechildren = document.getElementById('synced-libraries-rows');
+		if (index >= treechildren.childNodes.length) {
+			return;
+		}
+		var row = treechildren.childNodes[index];
+		var cell = row.firstChild.firstChild;
+		cell.setAttribute('value', cell.getAttribute('value') != 'true');
+
+		return this.toggleSyncedLibrary(index);
+	},
+
+
+	clickSyncedLibrary: function (event) {
+		var tree = document.getElementById("synced-libraries-tree");
+		var row = {}, col = {}, child = {};
+		tree.treeBoxObject.getCellAt(event.clientX, event.clientY, row, col, child);
+		
+		if (col.value.element.id != 'synced-libraries-checked') {
+			return;
+		}
+		// if clicked on checkbox update pref
+		return this.toggleSyncedLibrary(row.value);
+	},
+	
+	
+	toggleSyncedLibrary: function (index) {
+		var treechildren = document.getElementById('synced-libraries-rows');
+		if (index >= treechildren.childNodes.length) {
+			return;
+		}
+		var row = treechildren.childNodes[index];
+		var id = parseInt(row.firstChild.childNodes[1].getAttribute('value'));
+		if (isNaN(id)) {
+			return;
+		}
+		
+		var syncedLibraries = JSON.parse(Zotero.Prefs.get('sync.syncedLibraries') || '[]');
+		var indexOfId = syncedLibraries.indexOf(id);
+		if (indexOfId != -1) {
+			syncedLibraries.splice(indexOfId, 1);
+		} else {
+			syncedLibraries.push(id);
+		}
+		Zotero.Prefs.set('sync.syncedLibraries', JSON.stringify(syncedLibraries));
+		 
+		var cell = row.firstChild.firstChild;
+		cell.setAttribute('value', indexOfId != -1);
+	},
+	
+	
+	initSyncedLibraries: Zotero.Promise.coroutine(function* () {
+		var apiKey = Zotero.Sync.Data.Local.getAPIKey();
+		var client = Zotero.Sync.Runner.getAPIClient({apiKey});
+		var keyInfo = yield Zotero.Sync.Runner.checkAccess(client, {timeout: 5000});
```

Comment:
What happens if this (or another API request) fails? Dialog should at least close.